### PR TITLE
[don't merge] Fix #8457, #8572

### LINKED ISSUE #8457: Dorothea, Vengeful Victim automatically gets removed from game once casted for its Disturb cost
image
As shown in the log, I cast [[Dorothea, Vengeful Victim]] from the graveyard with its Disturb cost, coming back as Dorothea's Retribution, but Dorothea just gets exiled as soon as it hits the battlefield.

### PR DESCRIPTION
**Pull request by awjackson**
This fixes Disturb cards that are Auras on their second side by copying the target and AttachEffect from the second side card's SpellAbility to the DisturbAbility.

If the second side is a creature rather than an Aura, its SpellAbility won't *have* any targets or effects, so I believe there's no need to explicitly check the card type of the second side card (there might not be any need to check that the Effect being copied is an AttachEffect either, but this way makes it a bit clearer why we're doing this)

### CODE REVIEW

File: Mage/src/main/java/mage/abilities/keyword/DisturbAbility.java
Comment (awjackson) [inline Mage/src/main/java/mage/abilities/keyword/DisturbAbility.java:55]: One thing I'm not sure about is whether it's necessary to defensively copy the Target and Effect objects that we're "borrowing" from the back side (i.e. `this.addTarget(target.copy())`). It seems to work without the defensive copies (I cast a Disturb card in aura mode, bounced it to my hand, got the card into my graveyard, and cast the same card again in aura mode, and everything seemed to go fine) but I don't know if it's actually correct or "safe".

File: Mage/src/main/java/mage/abilities/keyword/DisturbAbility.java
Comment (awjackson) [inline Mage/src/main/java/mage/abilities/keyword/DisturbAbility.java:58]: One thing I'm not sure about is whether it's necessary to defensively copy the Target and Effect objects that we're "borrowing" from the back side (i.e. this.addTarget(target.copy())). It seems to work without the defensive copies (I cast a Disturb card in aura mode, bounced it to my hand, got the card into my graveyard, and cast the same card again in aura mode, and everything seemed to go fine) but I don't know if it's actually correct or "safe".

**Comment by jeffwadsworth** [comment]
The implementation has core issues.  Note the "spellAbilityToResolve" is never referenced.

**Review by JayDi85 — CHANGES_REQUESTED** [review]
I don't like the idea to add a custom copy code of targets and effects -- original copy spell code must do all work under the hood.

Must be researched again for original of the bug and another solutions -- I'll do it later.

**Comment by awjackson** [comment]
I've been researching how Auras work in xmage and I'm confident that this is the only reasonable way to handle Disturb Auras.

>115.1b Aura spells are always targeted. An Aura’s target is specified by its enchant keyword ability (see rule 702.5, “Enchant”). The target is chosen as the spell is cast; see rule 601.2c. An Aura permanent doesn’t target anything; only the spell is targeted.

In xmage, Auras' uniqueness is reflected in that they are the only permanents whose `SpellAbility` has targets and has an effect (`AttachEffect`). When we cast a Disturb card that is going to become an Aura, we need a `SpellAbility` that has those things.

Furthermore, we need to know about the back side's targets *before* the player starts casting the spell, See rule 712.7a:

>712.7a Only the face that will be face up on the stack is evaluated to determine if it can be cast. Only that face is considered to be put onto the stack.

A Disturb card in a graveyard shouldn't be castable if its back face is an Aura and there aren't any legal targets for that Aura. It's too late to replace or modify the SpellAbility during the casting process.

**Comment by awjackson** [comment]
Is this way better? Now we construct the ability as a copy of the back face's SpellAbility (using a helper method added to CardImpl, since super() must be the first line in a constructor), rather than as a copy of the front face's SpellAbility.

**Comment by jeffwadsworth** [comment]
That looks much better, but JayDi would know for sure.

**Comment by awjackson** [comment]
@JayDi85 I'm going to merge this tomorrow unless you come out and explain why you think it's still wrong. We're dealing with an entire keyword mechanic from the second-most-recent set that just doesn't work at all, and it's been sat on with no progress since November.

**Comment by JayDi85** [comment]
Don’t merge until new release. New features/fixes on the freeze.

**Comment by jeffwadsworth** [comment]
Since new releases are held up for a bit, perhaps this commit can be applied.

**Comment by Alex-Vasile** [comment]
Closes #8901, closes #8924

**Comment by Alex-Vasile** [comment]
> Since new releases are held up for a bit, perhaps this commit can be applied.

While we're waiting, do you have the time to add a test for casting them as auras?

**Comment by jeffwadsworth** [comment]
> > Since new releases are held up for a bit, perhaps this commit can be applied.
> 
> While we're waiting, do you have the time to add a test for casting them as auras?

Just give a specific example for the test.

**Comment by Alex-Vasile** [comment]
> > > Since new releases are held up for a bit, perhaps this commit can be applied.
> 
> > 
> 
> > While we're waiting, do you have the time to add a test for casting them as auras?
> 
> 
> 
> Just give a specific example for the test.

Ghastly Mimicry onto a creature, see that it attaches and then see that a clone gets made?

**Comment by jeffwadsworth** [comment]
> Ghastly Mimicry

Well, I know the test will work fine with this PR.  The issue was pretty simple and I really have no idea why this is still in here.
Once this is merged, I will add a test for it.

**Comment by Alex-Vasile** [comment]
@JayDi85 Can this be merged? It's been waiting for a while and fixes a bug that renders several cards unusable.

**Pull request merged**
